Aydjay branch

aydjay opened the pull request:
Been wanting to make some changes to procurement so I got the code and saw all the compiler warnings. This PR will reduce the number of compiler warnings the solution gives

Stickymaddness commented:
This looks great! Moving to Json.Net has been on my to-do list for a long time!

I currently have very little time to dedicate to Procurement, but as soon as I have some time available I will look at this PR as a priority.

Thanks for the contribution!

aydjay commented:
No problem, I have added unit tests and moved over some key calls to using JSon.Net.

Moving the get stash over is quite tricky as there are alot of object factory calls that are made. Not sure what is the best approach to improve this part of the code base.

There is a unit test and MOQ implemented for the rework but it's not a small job.

Stickymaddness commented on POEApi.Model.Tests/PoeModelTests.cs:
```diff
@@ -0,0 +1,97 @@
+    [TestClass]
+    public class PoeModelTests
+    {
+        private Mock<ITransport> _mockTransport;
```
Nitpick, but please remove the underscores from private variables to maintain naming consistency throughout the app.

Stickymaddness commented on POEApi.Model/POEApi.Model.csproj:
```diff
@@ -1,126 +1,137 @@
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>POEApi.Model</RootNamespace>
+    <AssemblyName>POEApi.Model</AssemblyName>
+    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
```
Is there a requirement for upgrading to v4.6.1?

aydjay replied:
There isn't a prereq for moving to 4.6.1 I just want to use the new language features.

Stickymaddness commented:
I'd still like to merge this, but due to the newer releases of Procurement your PR now has a few conflicts.

aydjay commented:
I'll see if I can get some time to look at this again.

This is still only a partial move to using JSon.Net 

IIRC the Stash logic etc... was really tricky to move across easily. I just did the low hanging fruit.

aydjay commented:
Right I think I've done the correct thing - I'm new to this git/github business.

I was able to launch the application and login using the new (Old now) code that I wrote in january using the 1.9.9 release.

I could browse my tabs etc...

aydjay commented:
Any news on merging this PR @Stickymaddness ?

aydjay commented:
Bumping!

Stickymaddness commented:
Thanks for updating.

Please will you remove the /packages/ folder and gitignore it, there is no need for them to be checked in. 

I know you're new to git, but please try keep your PRs small and contained to one area that is being changed, it makes processing a PR considerably easier.

aydjay commented:
Thanks for getting back to me over this PR.

I made many mistakes in my approach to doing this work - but I have learnt! Thanks for your patience I have updated the gitignore file and "Cleansed" the repo of unwanted changes.
